<!-- Launches oculus_publisher, oculus_subscriber, and dynamic reconfigure gui
     with default sonar parameters (as defined in SimpleFireMessage.cpp)
-->
<launch>
  <arg name="range" default="2"/>
  <arg name="gainPercent" default="50"/>
  <arg name="gamma" default="127"/>
  <arg name="pingRate" default="0"/>
  <arg name="freqMode" default="2"/>
  <arg name="ipAddress" default="auto"/>
  <arg name="send_gain" default="true"/>
  <arg name="data_size" default="8bit"/>

  <!-- If true, also launches a sonar_draw node -->
  <arg name="sonar_draw" default="false"/>

  <!--  launch-prefix="gdb -ex run args" -->
  <node name="oculus" pkg="oculus_sonar_driver" type="oculus_driver" output="screen">
    <param name="ipAddress" value="$(arg ipAddress)"/>
    <param name="range" value="$(arg range)"/>
    <param name="pingRate" value="$(arg pingRate)"/>
    <param name="gamma" value="$(arg gamma)"/>
    <param name="gainPercent" value="$(arg gainPercent)"/>
    <param name="freqMode" value="$(arg freqMode)"/>
    <param name="data_size" value="$(arg data_size)"/>
    <param name="send_gain" value="$(arg send_gain"/>
  </node>

  <node pkg="oculus_sonar_driver" type="oculus_draw" name="oculus_draw" if="$(arg sonar_draw)"/>
  <!-- <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure"/> -->

</launch>
